drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -56,11 +60,24 @@ static inline void msg_done(struct device *dev, unsigned int flags)
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
 
+	if (!LL_I2C_IsEnabledAutoEndMode(i2c))
+	{
+		/* Wait for transfer to complete */
+		while (!LL_I2C_IsActiveFlag_TC(i2c) ){
+			;
+		}
+		/* Issue stop condition if necessary */
+		if ((flags & I2C_MSG_RESTART) == 0) {
+			LL_I2C_GenerateStopCondition(i2c);
+		}
+	}
+
+	/* Wait for stop condition on bus if necessary */
 	if ((flags & I2C_MSG_RESTART) == 0) {
```

Comment:
The two if ((flags & I2C_MSG_RESTART) == 0) should be merged

Reply:
well, this last if ((flags & I2C_MSG_RESTART) == 0) needs to be evaluated for both auto end and none auto end mode. So I don't see how they can be merged without breaking functionality or needing two if(!LL_I2C_IsEnabledAutoEndMode)